### PR TITLE
FieldRefAccess expansion

### PR DESCRIPTION
**Pull request by DenJur**
Added overload that takes FieldInfo as an argument. Also added support for static fields.
Not sure what behavior you want for a version that takes in a fieldname. For now I left it as it was in the original.

### CODE REVIEW

File: Harmony/Tools/AccessTools.cs [hunk 1]
```diff
@@ -733,23 +733,43 @@ public static object[] ActualParameters(MethodBase method, object[] inputs)
 											BindingFlags.DeclaredOnly;
 
 			var fi = typeof(T).GetField(fieldName, bf);
-			if (fi == null)
-				throw new MissingFieldException(typeof(T).Name, fieldName);
```
Comment (pardeike) [inline Harmony/Tools/AccessTools.cs]: Why is this being removed?

File: HarmonyTests/Tools/TestAccessTools.cs [hunk 2]
```diff
@@ -169,5 +169,80 @@ public void AccessTools_TypeExtension_Types()
...
+	  }
+
+	  [TestMethod]
+		public void AccessTools_FieldRefAccess_ByFieldInfo_Static()
+		{
+			var fieldInfo = typeof(AccessToolsClass).GetField("field3", BindingFlags.Static | BindingFlags.NonPublic);
+			Assert.IsNotNull(fieldInfo);
+			var instance = new AccessToolsClass();
+			var fieldRef = AccessTools.FieldRefAccess<AccessToolsClass, String>(fieldInfo);
+			ref String value = ref fieldRef(instance);
```
Comment (pardeike) [inline HarmonyTests/Tools/TestAccessTools.cs]: Passing an instance makes no sense for a static field

File: Harmony/Tools/AccessTools.cs (hunk 1 above)
Comment (DenJur) [reply Harmony/Tools/AccessTools.cs]: It is checked later

**Comment by pardeike** [comment]
See my comments. Maybe an overload for static fields would be better?

**Pull request merged**